### PR TITLE
Added metadata to configure remove-live

### PR DESCRIPTION
**Pull request by wachterjohannes**
| Q | A
| --- | ---
| Bug fix? | no
| New feature? | yes
| BC breaks? | no
| Deprecations? | no
| Fixed tickets | none
| Related issues/PRs | requires https://github.com/sulu/sulu-document-manager/pull/109 and related to https://github.com/sulu/SuluArticleBundle/pull/139 build ontop https://github.com/sulu/sulu/pull/3299
| License | MIT
| Documentation PR | none

#### What's in this PR?

This PR adds a check which uses a new metadata property to avoid removing the live node for special documents.

#### To Do

- [x] Tests

### CODE REVIEW

File: src/Sulu/Bundle/ContentBundle/Document/Subscriber/PublishSubscriber.php [hunk 1]
```diff
@@ -111,6 +119,11 @@ public function createNodeInPublicWorkspace(PersistEvent $event)
      */
     public function removeNodeFromPublicWorkspace(RemoveEvent $event)
     {
+        $metadata = $this->metadataFactory->getMetadataForClass(get_class($event->getDocument()));
```
Comment (danrot) [inline src/Sulu/Bundle/ContentBundle/Document/Subscriber/PublishSubscriber.php]: Extract `$event->getDocument()` into own variable, since it is used again afterwards.

**Comment by wachterjohannes** [comment]
@danrot because i have to work on this branches for the article-bundle

**Comment by wachterjohannes** [comment]
@danrot should be fixed now

**Comment by danrot** [comment]
Still needs a change because of https://github.com/sulu/sulu-document-manager/pull/109#discussion_r111170981

**Comment by wachterjohannes** [comment]
@danrot done

File: composer.json [hunk 2]
```diff
@@ -28,7 +28,7 @@
         "imagine/imagine": "~0.6.1",
         "massive/search-bundle": "0.16.*",
         "sensio/framework-extra-bundle": "^3.0",
-        "sulu/document-manager": "dev-develop",
+        "sulu/document-manager": "dev-enhancement/remove-live-metadata",
```
Comment (danrot) [inline composer.json]: I've merged the other PR, can you undo this change?

**Comment by wachterjohannes** [comment]
@danrot done

**Comment by danrot** [comment]
Tests are failing.
